tests/perf/http_client_perf.cc

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2024 ScyllaDB
+ */
+
+#include <seastar/core/seastar.hh>
+#include <seastar/core/app-template.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/http/client.hh>
+#include <seastar/http/request.hh>
+#include <../../tests/unit/loopback_socket.hh>
+#include <fmt/printf.h>
+#include <string>
+
+using namespace seastar;
+using namespace std::chrono_literals;
+
+class server {
+    seastar::server_socket _ss;
+    seastar::connected_socket _cs;
+    seastar::input_stream<char> _in;
+    seastar::output_stream<char> _out;
+    sstring _req;
+
+    future<> do_echo() {
+        return _in.read().then([this] (temporary_buffer<char> buf) {
+            if (buf.empty()) {
+                return make_ready_future<>();
+            }
+
+            _req += sstring(buf.get(), buf.size());
+            if (_req.ends_with("\r\n\r\n")) {
+                sstring r200("HTTP/1.1 200 OK\r\nHost: test\r\n\r\n");
+                return _out.write(r200).then([this] {
+                    return _out.flush().then([this] {
+                        _req = "";
+                        return do_echo();
```

Comment:
There is the theoretical and extremely unlikely possibility that if in.read() is ready many times in sequence and _out.write() is also ready many times, this code will have very deep recursion and crash. But since this is just a test and since it's so unlikely that this can even happen, I think it's fine.

Maybe everything will be nicer if just written using coroutines and a normal endless loop?

Reply:
Not only out.write() should be ready, but also out.flush(). The way this test runs (ping-pong mode) it _must_ switch to client fiber after flush(), as client doesn't pipeline requests and generates the new one after it sees response from the server

> Maybe everything will be nicer if just written using coroutines and a normal endless loop?

Nicer, yes, definitely. Also client-part looks nicer with coroutines